00X-ABI-Stable-Module-API.md

```diff
@@ -0,0 +1,369 @@
+| Title  | ABI Stable Module API                 |
+|--------|---------------------------------------|
+| Author | @mhdawson, @stefanbu, @Ianwjhalliday  |
+| Status | DRAFT                                 |
+| Date   | 2016-08-11                            |
+
+# Introduction
+
+The module ecosystem is one of the features making Node.js the fastest
+growing runtime.  An important subset of modules are those which have a
+native component. In specific cases it is important to be able to leverage native
+code in order to be able to:
+
+* use existing third party components written in C/C++
+* access physical devices, for example a serial port
+* expose functionality from the operating system not otherwise available
+* achieve higher speed than possible in native JavaScript
```

Comment:
Might also be worth noting the possibility of pulling CPU-intensive work out-of-band from the JS thread. That's sort of "speed" related, but maybe different enough that it warrants an explicit mention.